tests/main/nvidia-files/task.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+set -xeu
+case "$1" in
+prepare)
+	# Install Nvidia userspace libraries at the designated version.
+	apt-get install -y \
+		libnvidia-common-"$PACKAGE_VERSION" \
+		libnvidia-compute-"$PACKAGE_VERSION":amd64 \
+		libnvidia-compute-"$PACKAGE_VERSION":i386 \
+		libnvidia-decode-"$PACKAGE_VERSION":amd64 \
+		libnvidia-decode-"$PACKAGE_VERSION":i386 \
+		libnvidia-encode-"$PACKAGE_VERSION":amd64 \
+		libnvidia-encode-"$PACKAGE_VERSION":i386 \
+		libnvidia-fbc1-"$PACKAGE_VERSION":amd64 \
+		libnvidia-fbc1-"$PACKAGE_VERSION":i386 \
+		libnvidia-gl-"$PACKAGE_VERSION":amd64 \
+		libnvidia-gl-"$PACKAGE_VERSION":i386
+
+	tests.cleanup defer apt-get remove --purge -y \
+		libnvidia-common-"$PACKAGE_VERSION" \
+		libnvidia-compute-"$PACKAGE_VERSION":amd64 \
+		libnvidia-compute-"$PACKAGE_VERSION":i386 \
+		libnvidia-decode-"$PACKAGE_VERSION":amd64 \
+		libnvidia-decode-"$PACKAGE_VERSION":i386 \
+		libnvidia-encode-"$PACKAGE_VERSION":amd64 \
+		libnvidia-encode-"$PACKAGE_VERSION":i386 \
+		libnvidia-fbc1-"$PACKAGE_VERSION":amd64 \
+		libnvidia-fbc1-"$PACKAGE_VERSION":i386 \
+		libnvidia-gl-"$PACKAGE_VERSION":amd64 \
+		libnvidia-gl-"$PACKAGE_VERSION":i386
+
+	# Look at the canary file libnvidia-glcore.so.* to get the exact version of
+	# the driver. This file is also used by snap-confine, as a pre-condition
+	# that the libraries are installed.
+	DRIVER_VERSION="$(find /usr/lib/x86_64-linux-gnu/ -name 'libnvidia-glcore.so.*' | sed -e 's,.*/libnvidia-glcore\.so\.,,')"
+
+	# Pretend we have Nvidia kernel module loaded, so that snap-confine enables
+	# special logic. The actual version we pretend to have is set later, as it
+	# must match installed libraries so that the right canary file is detected by
+	# snap-confine.
+	mkdir -p /tmp/sys-module/nvidia
+	echo "$DRIVER_VERSION" >/tmp/sys-module/nvidia/version
+	if os.query is-xenial || os.query is-bionic; then
```

Comment:
This is somewhat cargo-culted from the other nvidia test but I will need it for the bionic extension to this test.